src/function/gds/page_rank.cpp

```diff
@@ -133,6 +133,37 @@ class PageRankAuxiliaryState : public GDSAuxiliaryState {
     PValues& pNext;
 };
 
+class KUZU_API PageRankFrontierPair : public FrontierPair {
+public:
+    PageRankFrontierPair(std::shared_ptr<PathLengths> curFrontier,
+        std::shared_ptr<PathLengths> nextFrontier)
+        : FrontierPair{curFrontier, nextFrontier} {}
+
+    void pageRankPinCurrFrontier(common::table_id_t tableID) {
+        curSparseFrontier->pinTableID(tableID);
+        curDenseFrontier->pinCurFrontierTableID(tableID);
+        // In PNextUpdateEdgeCompute, addNodesToNextDenseFrontier is boundID
+        // should pin fromTableID in there.
+        nextDenseFrontier->pinNextFrontierTableID(tableID);
```

Comment:
Hi @excaliburwyj, other changes looks fine to me. But I don't understand this line

If I'm extending from person -> organization. Next dense frontier you are pinning for `person` instead of `organization`. Why is this the case?